From 29870e6f38fef098cbd13a59969e5a5f0ad77985 Mon Sep 17 00:00:00 2001
From: jakub <jakub@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Fri, 20 May 2011 18:23:36 +0000
Subject: [PATCH] 	PR bootstrap/49086
 	* gimple-fold.c (and_comparisons_1, or_comparisons_1): Return NULL
 	for PHI args that are SSA_NAME_IS_DEFAULT_DEF.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_6-branch@173968 138bc75d-0d04-0410-961f-82ee72b054a4

index 6ff6f81..910e3b0 100644
--- a/gcc/gimple-fold.c
+++ b/gcc/gimple-fold.c
@@ -2183,7 +2183,8 @@ and_comparisons_1 (enum tree_code code1, tree op1a, tree op1b,
 							code2, op2a, op2b))
 			return NULL_TREE;
 		    }
-		  else if (TREE_CODE (arg) == SSA_NAME)
+		  else if (TREE_CODE (arg) == SSA_NAME
+			   && !SSA_NAME_IS_DEFAULT_DEF (arg))
 		    {
 		      tree temp;
 		      gimple def_stmt = SSA_NAME_DEF_STMT (arg);
@@ -2644,7 +2645,8 @@ or_comparisons_1 (enum tree_code code1, tree op1a, tree op1b,
 							code2, op2a, op2b))
 			return NULL_TREE;
 		    }
-		  else if (TREE_CODE (arg) == SSA_NAME)
+		  else if (TREE_CODE (arg) == SSA_NAME
+			   && !SSA_NAME_IS_DEFAULT_DEF (arg))
 		    {
 		      tree temp;
 		      gimple def_stmt = SSA_NAME_DEF_STMT (arg);
-- 
1.7.0.4

